plugin/trino-ldap-group-provider/src/main/java/io/trino/plugin/ldapgroup/LdapGroupProviderSimpleClient.java

```diff
@@ -0,0 +1,160 @@
+                        .withSearchFilter(userSearchFilter)
+                        .withFilterArguments(user)
+                        .build(),
+                search -> {
+                    if (!search.hasMore()) {
+                        log.debug("LDAP search for user [%s] using filter pattern [%s] found no matches", user, userSearchFilter);
+                        return Optional.empty();
+                    }
+                    SearchResult result = search.next();
+                    ImmutableMap.Builder<String, List<String>> resAttributesBuilder = ImmutableMap.builder();
+                    for (String attributeName : userSearchAttributes) {
+                        Attribute attribute = result.getAttributes().get(attributeName);
+                        if (attribute == null) {
+                            log.warn("User [%s] does not have user search attribute [%s]", user, attributeName);
+                            continue;
+                        }
+                        List<String> values = Streams.stream(attribute.getAll().asIterator())
+                                .map(Object::toString)
+                                .collect(toImmutableList());
+                        if (values.isEmpty()) {
+                            log.debug("User [%s] has attribute [%s] with not values", user, attributeName);
+                            continue;
+                        }
+                        resAttributesBuilder.put(attributeName, values);
+                    }
+                    return Optional.of(new LdapSearchUserResult(result.getNameInNamespace(), resAttributesBuilder.buildOrThrow()));
+                });
+    }
+
+    /**
+     * Get user groups from values of memberOfAttribute.
+     *
+     * @param searchUserResult User search result from {@link LdapGroupProviderClient#getUser}.
+     * @return A list of user groups.
+     */
+    @Override
+    public List<String> getGroups(LdapSearchUserResult searchUserResult)
+    {
+        return searchUserResult
+            .userAttributes()
+            .getOrDefault(userMemberOfAttribute, ImmutableList.of())
+            .stream()
+            .map(this::getGroupRelativeDN)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(toImmutableList());
+    }
+
+    private Optional<String> getGroupRelativeDN(String groupDN)
+    {
+        LdapName groupLdapName;
+        try {
+            groupLdapName = new LdapName(groupDN);
+        }
+        catch (NamingException namingException) {
+            log.warn(namingException, "Group has malformed DN [%s]. Ignoring group.", groupDN);
+            return Optional.empty();
+        }
+        List<Rdn> groupRDNs = groupLdapName.getRdns();
+        if (groupRDNs.isEmpty()) {
+            log.warn("Group with DN [%s] has no Relative Domain Names. Ignoring group.", groupDN);
+            return Optional.empty();
+        }
+        Rdn lastRDN = groupRDNs.getLast();
```

Comment:
an LDAP Distinguished Name (DN) is a chain of multiple Relative Distinguished Names (RDNs). for example in `cn=users,ou=groups,dc=trino,dc=io` - `cn=users` and `ou=groups` are RDNs.

these are meant to be read from right to left (you can notice they get more specific). `getRdns()` will return them in this order, so the last element is the least specific RDN, usually what we would call the "name". in the example above it would be `cn=users`.

parsing group names from DNs is an heuristic based on the assumption that the last RDN is the Canonical Name we are looking for.